packages/grafana-ui/src/components/Graph/GraphLegendItem.tsx

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import { css } from 'emotion';
+import { LegendSeriesIcon } from '../Legend/LegendSeriesIcon';
+import { LegendItem } from '../Legend/Legend';
+
+interface GraphLegendItemProps {
+  item: LegendItem;
+  onLabelClick: (item: LegendItem, event: React.MouseEvent<HTMLDivElement>) => void;
+  onSeriesColorChange: (color: string) => void;
+  onToggleAxis: () => void;
+}
+
+export const GraphLegendItem: React.FunctionComponent<GraphLegendItemProps> = ({
+  item,
+  onSeriesColorChange,
+  onToggleAxis,
+  onLabelClick,
+}) => {
+  return (
+    <>
+      <LegendSeriesIcon color={item.color} onColorChange={onSeriesColorChange} onToggleAxis={onToggleAxis} />
+      <div
+        onClick={event => onLabelClick(item, event)}
+        className={css`
+          cursor: pointer;
+        `}
```

Comment:
Rather than using strings, should we use:
```
            className={css({
              cursor: 'pointer'
            })}
```
I only say that since I copied this example... and found that using strings does not have any type checking for the style you put in

Reply:
There are pros and cons of this approach. 

I personally prefer using template strings as they create one semantic space for interpolation  as opposed to the object approach. Then, when using css`` you are just writing css code so you i.e don't need additional quotation marks. It's also harder(honestly speaking I'm not sure if its' even possible) to use pseudo selectors when using objects.

And finally, I don't see `css(...)` supporting typecheck on a single property level. I've just tested it and didn't get any type checking.

Btw have you tried https://github.com/styled-components/vscode-styled-components @ryantxu ?

Reply:
Not sure why I assumed you use VSC:) There is a vim plugin https://github.com/styled-components/vim-styled-components and WebStorm's as well: https://github.com/styled-components/webstorm-styled-components

Reply:
nice -- yes I use vscode :)

By type checking I mean this:
![image](https://user-images.githubusercontent.com/705951/55894100-5d381f00-5b6e-11e9-9b20-30f1a84bacfe.png)

vs

![image](https://user-images.githubusercontent.com/705951/55894121-6b863b00-5b6e-11e9-9602-58f46f0f2716.png)

but yes -- lots of cases where you just have to do it properly